lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -167,7 +167,21 @@ def write(chunk)
           'entries' => [],
         }
         arr.each do |time, record|
-          if (record.has_key?('timeNanos'))
+          if (record.has_key?('timestamp') &&
+              record['timestamp'].has_key?('seconds') &&
+              record['timestamp'].has_key?('nanos'))
+            ts_secs = record['timestamp']['seconds']
+            ts_nanos = record['timestamp']['nanos']
+            record.delete('timestamp')
+          elsif (record.has_key?('timestampSeconds') &&
+                 record.has_key?('timestampNanos'))
```

Comment:
One other thing I did debate was adding support for millis/micros since the user might have a log format that uses those and we don't have an opportunity to do math. I decided not to for simplicity but I could be convinced...